profiler/src/main/java/com/splunk/opentelemetry/profiler/RecordingEscapeHatch.java

```diff
@@ -16,9 +16,148 @@
 
 package com.splunk.opentelemetry.profiler;
 
+import java.io.IOException;
+import java.nio.file.FileStore;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 class RecordingEscapeHatch {
 
+  private static final long MIN_FREE_SPACE_BYTES = 100 * 1024 * 1024; // 100MiB
+  private static final Logger logger = LoggerFactory.getLogger(RecordingEscapeHatch.class);
+  private static final Duration MAX_PENDING_DURATION = Duration.ofMinutes(5);
+
+  private final RecordingFileNamingConvention namingConvention;
+  private final boolean configKeepsFilesOnDisk;
+  private final long maxFileCount;
+  private final FilesShim filesShim;
+
+  RecordingEscapeHatch(Builder builder) {
+    this.namingConvention = builder.namingConvention;
+    this.configKeepsFilesOnDisk = builder.configKeepsFilesOnDisk;
+    this.filesShim = builder.filesShim;
+    this.maxFileCount = MAX_PENDING_DURATION.toMillis() / builder.recordingDuration.toMillis();
+  }
+
   public boolean jfrCanContinue() {
-    return true;
+    boolean result = freeSpaceIsOk() && notFileBacklogged();
+    if (!result) {
+      logger.warn("** THIS WILL RESULT IN LOSS OF PROFILING DATA **");
+    }
+    return result;
+  }
+
+  private boolean freeSpaceIsOk() {
+    try {
+      Path outputPath = namingConvention.getOutputPath();
+      FileStore store = filesShim.getFileStore(outputPath);
+      long usableSpace = store.getUsableSpace();
```

Comment:
Any idea what these APIs do on docker? Is it going to be returning something from the host, or is it for-sure just the container?

Reply:
I would expect them to do the same thing in a container.  Containers have a certain amount of space available to them, and volume mounts could have been created to map onto the host system.